### ISSUE TITLE
Enable Choice objects in molecules

### ISSUE THREAD

**Issue by fgrunewald** [marrink-lab/vermouth-martinize#296]
Currently only links can have choices (i.e. a link might apply to a molecule if the respective attribute matches one in the link choice list). However, it would be an interesting feature to have also Choice objects as molecule attributes used in link matching. For example I am trying to write links that match a respective functional group. However, functional groups can overlap. An atom can be part of two functional groups. In this case all links that have the functional group attribute matching the one in the choice list, should be matched. 

https://github.com/marrink-lab/vermouth-martinize/blob/bd58a5143e81dd2cd35584d7de2a0050ff3e7d5e/vermouth/molecule.py#L1324

**Comment by pckroon** [comment]
I'm more inclined to add a "Contains" LinkPredicate. That fits better in the current code structure as well.
This would then match if `value in mol.nodes[node][attribute]` is True.
What do you think?

**Comment by fgrunewald** [comment]
Contains would also be fine, but remember we need to differentiate links and blocks when reading the ff file, then. That is easily doable. Would it also support having lists on both sides. I would interpret it as if any of the values given in molecule matches any of the values in link.

**Comment by pckroon** [comment]
> Contains would also be fine, but remember we need to differentiate links and blocks when reading the ff file, then.

This is already done, and will require to be done anyway since they result in fundamentally different objects.

> Would it also support having lists on both sides. I would interpret it as if any of the values given in molecule matches any of the values in link.

That would be something different I think. Then you're talking more about an intersection. I'm ok with having both operations.
Also, I don't think the current parser can really deal with lists in a nice way. See also #175 #176 #177

**Comment by fgrunewald** [comment]
I agree to make these two different objects and parse them accordingly. I'm fine with that.

In my opinion overall it would be highly beneficial for vermouth as package to have more freedom in utilizing these sort of list arguments for node attributes and have a way to parse them and also interpret them in DoLinks. Question is: Are we revisiting this with the parser or should me make a PR (perhaps as draft) to see how the nitty-gritty works out.
